dali/util/nvml.h

```diff
@@ -241,6 +241,43 @@ inline void Shutdown() {
   CUDA_CALL(nvmlShutdown());
 }
 
+
+class nvmlHandle {
```

Comment:
We have NvmlError and NvmlBadAlloc, so I guess it'd be good to have Nvml... something.
I'm not convinced we should call it NvmlHandle, since it's not a handle nor does it wrap any handle.
Other names to consider:
NvmlScope
NvmlInstance

Reply:
Good point. Fixed.